test/tools/unified-spec-runner/match.ts

```diff
@@ -323,6 +347,23 @@ export function specialCheck(
         ejson`expected value at path ${path.join('')} NOT to exist, but received ${actual}`
       ).to.be.false;
     }
+  } else if (isMatchAsDocumentOperator(expected)) {
+    if (typeof actual === 'string') {
+      const actualDoc = EJSON.parse(actual, { relaxed: false });
+      resultCheck(actualDoc, expected.$$matchAsDocument as any, entities, path, true);
+    } else {
+      expect.fail(`Expected value at path ${path.join('')} to be string, but received ${actual}`);
+    }
+  } else if (isMatchAsRootOperator(expected)) {
+    expect(
+      typeof actual,
+      `Expected value at path ${path.join('')} to be object, but received ${actual}`
+    ).to.equal('object');
+    expect(typeof expected.$$matchAsRoot, 'Value of $$matchAsRoot must be an object').to.equal(
```

Comment:
Spec states that `$$matchAsDocument` should only be used when `expected` and `actual` are documents